internal/juju/offers.go

```diff
@@ -390,3 +401,53 @@ func (c offersClient) RemoveRemoteOffer(input *RemoveRemoteOfferInput) []error {
 
 	return nil
 }
+
+// GrantOffer adds access to an offer managed by the access offer resource.
+// No action or error if the access was already granted to the user.
+func (c offersClient) GrantOffer(input *GrantRevokeOfferInput) error {
+	conn, err := c.GetConnection(nil)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = conn.Close() }()
+
+	client := applicationoffers.NewClient(conn)
+
+	for _, user := range input.Users {
+		err = client.GrantOffer(user, input.Access, input.OfferURL)
+		if err != nil {
+			// ignore if user was already granted
+			if strings.Contains(err.Error(), "user already has") {
```

Comment:
i don't think the juju api client returns an error in that case... maybe @hmlanigan knows about this

Reply:
I got this error while testing and the python library does something similar:
https://github.com/juju/python-libjuju/blob/1907f7af91195712badc8358d405424fc85f3ee0/juju/controller.py#L698